packages/cog/src/cli/actions/action.cog.ts

```diff
@@ -11,6 +11,33 @@ import { buildCogForQuadKey, CogJob } from '../../cog/cog';
 import { buildWarpedVrt } from '../../cog/cog.vrt';
 import { getJobPath, makeTempFolder } from '../folder';
 
+export function extractYearFromName(name: string): number {
+    const re = /(?:^|\D)(\d{4})(?:$|\D)/g;
+
+    let year = 2000;
+
+    for (let m = re.exec(name); m != null; m = re.exec(name)) {
+        const t = +m[1];
+        if (t > year) year = t;
+    }
+
+    return year;
```

Comment:
I am not sure we want to default to year `2000`

some imagery sets will not have the year inside of them.

Reply:
What do you suggest if year is missing? 1900?

Reply:
maybe leave it empty, its upto the user to configure it. or even -1?

Reply:
-1 since the field is not optional